### DISCUSSION TITLE
2 New Properties That Should be Added to Buttons

### DISCUSSION THREAD

**Discussion by techprotom** [xamarin/Xamarin.Forms#12992] [Ideas]
### **Other platforms for creating applications such as Kotlin provide a much richer set of controls when compared to Xamarin.Forms.**

But of course Xamarin.Forms has one key feature which makes it stand out from the rest: the ability to create **multiplatform** applications in record time. Xamarin.Forms developers have done an amazing job on providing more options on the UI-side of things - but I do think that more time and effort needs to be invested in using XAML/MVU for creating the UI - instead of having to create custom renderers all of the time.

Crucial properties on components which will be used tens of thousands of times (such as buttons) are missing key properties that if added could make Xamarin.Forms that much better to use:

**Proposal 1 - the ability to set uppercase and lowercase on buttons**

Right now - I am creating a custom renderer just to make it so button text is displayed as lower-case. 

Xamarin.Forms should consider adding a property titled 'IsUpperCase' on each button. The user will have the option to set this to either true or false. Realistically - this should not be a massive problem to implement. Sentence case is not my preferred way of creating buttons (not always) - so this can open the eyes up of many UI designers such as me.

_Example_

`<Button Text="Hello"  IsUpperCase="False"/>`

This shouldn't be a problem to implement with bindable properties.

**Propsal 2 - the ability to add shadows to buttons/other components**

Shadows can add depth to the user interface - here is an example for creating a shadow property on a button:

`<Button Text="Hello" IsUpperCase="False">
<Button.Effects><DropShadowEffect X="5" Y="3" Shadow="15px"> </Button.Effects></Button>`

The implementation could be similar to that of a UWP/WPF button.

I am just adding 2 properties for now - but if anyone has anymore ideas please tell me below.

What do you guys think about this?

![image](https://user-images.githubusercontent.com/72473828/100561095-b729c180-331c-11eb-9e6a-90c31495a944.png)

**Comment by pictos** [comment]
Hey @techprotom, I don't if you are aware, but we have the `TextTransform` prop on buttons, that allow you to make your text all upper case or not. 
`button.TextTransform = TextTransform.Uppercase`.

**Reply by techprotom** [reply]
Hello pictos - I appreciate you being on this discussion with me. Yes I've heard about this and it's good that you are out with new properties for buttons already!

I would also hope you could consider the ability to add drop shadow to buttons/other components using a dropshadoweffect. I've been developing for WPF for around 4 years and WPF XAML is more feature-rich. I would love it if for a start you could add the ability to create a dropshadow for a button to more finely tune your UI and add more depth to flat controls.

Xamarin.Forms is such a great framework and more options for the UI are added each day which I appreciate very much!

I also like your blogs and I appreciate the work you are putting in Xamarin.Forms.

**Comment by Cfun1** [comment]
For shadow effect: https://github.com/xamarin/XamarinCommunityToolkit/issues/461
